src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,228 @@
+                    + car + " in cars DB.", throwable);
+        }
+        removeDriversRelations(car);
+        List<Driver> drivers = car.getDrivers();
+        if (drivers != null) {
+            for (Driver driver : drivers) {
+                addDriverRelationForCar(car, driver);
+            }
+        }
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long carId) {
+        String query = "UPDATE cars SET deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, carId);
+            int numberOfDeletedRows = statement.executeUpdate();
+            return numberOfDeletedRows != 0;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't delete car with id " + carId, throwable);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getAllByDriverRequest = "SELECT c.name as car_name, c.id as car_id, "
+                + "m.name as manufacturer_name, m.country as manufacturer_country, "
+                + "m.id as manufacturer_id  from cars_drivers cd "
+                + "LEFT JOIN cars c ON cd.car_id = c.id "
+                + "LEFT JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE driver_id = ? AND c.deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarsByDriverStatement
+                        = connection.prepareStatement(getAllByDriverRequest)) {
+            getCarsByDriverStatement.setLong(1, driverId);
+            ResultSet resultSet = getCarsByDriverStatement.executeQuery();
+            while (resultSet.next()) {
+                Car car = getCarWithManufacturerFromResultSet(resultSet);
+                cars.add(car);
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't get cars by drivers id " + driverId,
+                    throwable);
+        }
+        parseDriversToListOfCars(cars);
+        return cars;
+    }
+
+    private void removeDriversRelations(Car car) {
+        String deleteRequest = "DELETE FROM cars_drivers WHERE car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement removeStatement = connection.prepareStatement(deleteRequest)) {
+            removeStatement.setLong(1, car.getId());
+            removeStatement.executeUpdate();
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't delete drivers dependencies for car with id "
+                    + car.getId(), throwable);
+        }
+    }
+
+    private void addDriverRelationForCar(Car car, Driver driver) {
```

Comment:
```suggestion
    private void addDriverRelationForCar(Car car) {
```
Since car has list of drivers we can pass only 1 param

Reply:
done